crates/db_schema/src/schema_setup/triggers.sql

```diff
@@ -0,0 +1,472 @@
+-- A trigger is associated with a table instead of a schema, so they can't be in the `r` schema. This is
+-- okay if the function specified after `EXECUTE FUNCTION` is in `r`, since dropping the function drops the trigger.
+--
+-- Tables that are updated by triggers should not have foreign keys that aren't set to `INITIALLY DEFERRED`
+-- (even if only other columns are updated) because triggers can run after the deletion of referenced rows and
+-- before the automatic deletion of the row that references it. This is not a problem for insert or delete.
+--
+--
+--
+-- Create triggers for both post and comments
```

Comment:
So many of these triggers are re-worked, that I hope our federation / unit tests are able to make sure they're all working correctly.

I'll also test this with production data locally, to make sure everything seems okay.